tests/src/utils/fs.test.ts

```diff
@@ -70,69 +31,118 @@ describe('normalizePath', () => {
 
 	test('should return null with invalid json file', async () => {
-		await expect(readJsonFile('.invalid-config.json')).resolves.toEqual(null);
+		mockFs({
+			'invalid.json': 'invalid-file',
+		});
+		await expect(readJsonFile('invalid.json')).resolves.toEqual(null);
+		mockFs.restore();
 	});
 });
 
 describe('validateFile', () => {
+	beforeAll(() => {
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('valid file returns true', async () => {
 		await expect(
-			validateFile('validTest/functions/index.func/index.js')
+			validateFile('functions/index.func/index.js')
 		).resolves.toEqual(true);
 	});
 
 	test('valid directory returns false', async () => {
-		await expect(
-			validateFile('validTest/functions/index.func')
-		).resolves.toEqual(false);
+		await expect(validateFile('functions/index.func')).resolves.toEqual(false);
 	});
 
 	test('invalid path returns false', async () => {
-		await expect(validateFile('invalidTest/invalidPath')).resolves.toEqual(
-			false
-		);
+		await expect(
+			validateFile('functions/invalid-index.func/index.js')
+		).resolves.toEqual(false);
 	});
 });
 
 describe('validateDir', () => {
+	beforeAll(() => {
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('valid directory returns true', async () => {
-		await expect(
-			validateDir('validTest/functions/index.func')
-		).resolves.toEqual(true);
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+		await expect(validateDir('functions/index.func')).resolves.toEqual(true);
+		mockFs.restore();
 	});
 
 	test('valid file returns false', async () => {
-		await expect(
-			validateDir('validTest/functions/index.func/index.js')
-		).resolves.toEqual(false);
+		await expect(validateDir('functions/index.func/index.js')).resolves.toEqual(
+			false
+		);
 	});
 
 	test('invalid path returns false', async () => {
-		await expect(validateDir('invalidTest/invalidPath')).resolves.toEqual(
-			false
-		);
+		await expect(validateDir('invalidPath')).resolves.toEqual(false);
 	});
 });
 
 describe('readPathsRecursively', () => {
+	beforeAll(() => {
+		mockFs({
+			root: {
+				functions: {
+					'(route-group)': {
+						'page.func': {
+							'index.js': 'page-js-code',
+						},
+					},
+					'index.func': {
+						'index.js': 'index-js-code',
+					},
+					'home.func': {
+						'index.js': 'home-js-code',
+					},
+				},
+			},
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('should read all paths recursively', async () => {
-		await expect(readPathsRecursively('validTest/functions')).resolves.toEqual([
-			'validTest/functions/(route-group)/page.func/index.js',
-			'validTest/functions/index.func/index.js',
-			'validTest/functions/home.func/index.js',
-		]);
+		const paths = await readPathsRecursively('root/functions');
+		expect(paths.length).toBe(3);
```

Comment:
```suggestion
		expect(paths.length).toEqual(3);
```
consistency 🙂

Reply:
from what I can tell we have a mix of both, I'm accepting the suggestion but if we want consistency we might need to change things anyways

Reply:
> from what I can tell we have a mix of both, I'm accepting the suggestion but if we want consistency we might need to change things anyways

Yeah, I was just looking in terms of the files changed in this PR - they all use `toEqual` and then there was this one straggler. The other couple of uses of `toBe` outside of this could do with changing at some point.

Reply:
do you prefer `toEqual` in general?

because I think that if values are simply built-in types then `toBe` is more appropriate and `toEqual` is for more complex types such as objects and arrays

seems kind of more correct to me but I doubt that there is any real difference between the two (even performance wise)

Reply:
> do you prefer `toEqual` in general?
> 
> because I think that if values are simply built-in types then `toBe` is more appropriate and `toEqual` is for more complex types such as objects and arrays
> 
> seems kind of more correct to me but I doubt that there is any real difference between the two (even performance wise)

I generally default to it, yeah - it's just personal preference at the end of the day. I don't really mind which we use, it's just that searching in VS Code, it looks like `toBe` is used a total of 4 times outside of the 2 you previously had in this PR. 🤷🏻‍♂️

Reply:
yeah as I said I _think_ `toBe` is more appropriate for simple types and `toEqual` for more complex ones, but I'm not really sure if that makes much of a difference (mostly this for me stems from the fact that if all can be done with `toEqual` why is `toBe` also provided in the first place? 😕)